File: 백준/Gold/3109. 빵집/빵집.py
import sys 
input = sys.stdin.readline 
sys.setrecursionlimit(10**6)

h, w = map(int, input().split())
graph = [[] for _ in range(h)]
for i in range(h):
    graph[i] = list(map(str, input().strip('\n')))

dx = [-1, 0, 1]
dy = [1, 1, 1]

def move(x, y):
    global cnt 
    graph[x][y]='a'
    if y==w-1:
        cnt+=1
        return True
    for i in range(3):
        nx = x+dx[i]
        ny = y+dy[i]
        if 0<=nx<h and 0<=ny<w and graph[nx][ny]=='.':
            if move(nx, ny):
                return True
cnt = 0
for i in range(h):
    move(i,0)

print(cnt)

File: 백준/Gold/3109. 빵집/빵집.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

var (
	reader    = bufio.NewReader(os.Stdin)
	writer    = bufio.NewWriter(os.Stdout)
	dx        = []int{-1, 0, 1}
	dy        = []int{1, 1, 1}
	cnt, h, w int
	graph     = [][]string{}
)

func move(x, y int) bool {
	graph[x][y] = "a"
	if y == w-1 {
		cnt++
		return true
	}
	for i := 0; i < 3; i++ {
		nx := x + dx[i]
		ny := y + dy[i]
		if 0 <= nx && nx < h && 0 < ny && ny < w && graph[nx][ny] == "." {
			if move(nx, ny) {
				return true
			}
		}
	}
	return false
}
func main() {
	defer writer.Flush()
	fmt.Fscanf(reader, "%d %d\n", &h, &w)
	var s string
	graph = make([][]string, h)
	for i := 0; i < h; i++ {
		fmt.Fscanf(reader, "%s\n", &s)
		graph[i] = strings.Split(s, "")
	}
	for i := 0; i < h; i++ {
		move(i, 0)
	}
	fmt.Fprintf(writer, "%d\n", cnt)
}
